ecp5/arch.cc

```diff
@@ -599,36 +597,349 @@ bool Arch::route()
 
 // -----------------------------------------------------------------------
 
+const float switchbox_x1 = 0.51;
+const float switchbox_x2 = 0.90;
+const float switchbox_y1 = 0.51;
+const float switchbox_y2 = 0.90;
+
+const float slice_x1 = 0.92;
+const float slice_x2 = 0.94;
+const float slice_y1 = 0.71;
+const float slice_y2 = 0.745 + 0.0068;
+const float slice_pitch = 0.0374 +0.0068;
+
+const float io_cell_v_x1 = 0.76;
+const float io_cell_v_x2 = 0.95;
+const float io_cell_v_y1 = 0.05;
+const float io_cell_v_y2 = 0.15;
+const float io_cell_v_pitch = 0.125;
+
+const float io_cell_h_x1 = 0.05;
+const float io_cell_h_x2 = 0.14;
+const float io_cell_h_y1 = 0.05;
+const float io_cell_h_y2 = 0.24;
+const float io_cell_h_pitch = 0.125;
+
 std::vector<GraphicElement> Arch::getDecalGraphics(DecalId decal) const
```

Comment:
Can you move the new larger gfx functions to a new gfx.cc rather than arch.cc? arch.cc is pretty slow to rebuild every time you change something already, and I'd prefer not to make that worse